src/daemon/emitter.cc

```diff
@@ -108,20 +110,30 @@ Emitter::Emitter(const proto::Configuration& configuration)
     }
   }
 
-  for (const auto& remote : config->emitter().remotes()) {
-    if (!remote.disabled()) {
+  if (runs_coordinators_task_) {
+    Vector<ResolveFn> resolvers;
+    resolvers.reserve(config->emitter().coordinators_size());
+    for (const auto& coordinator : config->emitter().coordinators()) {
+      if (coordinator.disabled())
+        continue;
       auto resolver = [
-        this, host = remote.host(), port = static_cast<ui16>(remote.port()),
-        ipv6 = remote.ipv6()
+        this,
+        host = coordinator.host(),
+        port = static_cast<ui16>(coordinator.port()),
+        ipv6 = coordinator.ipv6()
       ]() {
         auto optional = resolver_->Resolve(host, port, ipv6);
         DCHECK(optional);
         optional->Wait();
         return optional->GetValue();
       };
-      Worker worker = std::bind(&Emitter::DoRemoteExecute, this, _1, resolver);
-      workers_->AddWorker("Remote Execute Worker"_l, worker, remote.threads());
+      resolvers.push_back(resolver);
     }
+    Worker worker = std::bind(&Emitter::DoPoll, this, _1, resolvers);
+    workers_->AddWorker("Coordinator Poll Worker"_l, worker);
+  } else {
```

Comment:
By design we should at first consider the default remotes from config (if any), and only then try to use remotes from coordinators. It's better in case all coordinators are down, but we know some good remotes to fallback to.